cpp/splinepy/py/py_fitting.hpp

```diff
@@ -202,4 +202,81 @@ py::dict InterpolateSurface(py::array_t<double> points,
   return dict_spline;
 }
 
+py::dict ApproximateSurface(py::array_t<double> points,
+                            int num_points_u,
+                            int num_points_v,
+                            int size_u,
+                            int size_v,
+                            int degree_u,
+                            int degree_v,
+                            bool centripetal) {
+
+  // Extract input array info.
+  py::buffer_info p_buf = points.request();
+  double* p_buf_ptr = static_cast<double*>(p_buf.ptr);
+
+  // Prepare vars for interpolation
+  int num_points = p_buf.shape[0];
+  int surface_dim = p_buf.shape[1];
+  if (surface_dim < 2) {
+    splinepy::utils::PrintAndThrowError(
+        "Query points should be at least 2D for surface fitting.");
+  }
+  std::vector<double> knot_vector_u, knot_vector_v, control_points;
+
+  splinepy::fitting::ApproximateSurface(p_buf_ptr,     //
+                                        num_points_u,  //
+                                        num_points_v,  //
+                                        surface_dim,   //
+                                        degree_u,      //
+                                        degree_v,      //
+                                        size_u,        //
+                                        size_v,        //
+                                        centripetal,   //
+                                        knot_vector_u, //
+                                        knot_vector_v, //
+                                        control_points //
```

Comment:
I know you took this outline from a previous implementation, but once again, the control points could just expect a pointer to the points, as they are all assigned within the function. Here you could just put the pybind data and avoid a copy operation in the end.